extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -151,41 +155,47 @@ describe('ProjectsController: project controller operations', function (): void
 
 		it('Should be able to add folder with reserved name as long as not at project root', async function (): Promise<void> {
 			const folderName = 'folder1';
-			testContext.apiWrapper.reset();
-			testContext.apiWrapper.setup(x => x.showInputBox(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(folderName));
-			testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
+			const stub = sinon.stub(vscode.window, 'showInputBox').resolves(folderName);
 
-			const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
+			const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
 			const project = await testUtils.createTestProject(baselines.openProjectFileBaseline);
 			const projectRoot = new ProjectRootTreeItem(project);
 
 			// make sure it's ok to add these folders if they aren't where the reserved folders are at the root of the project
 			let node = projectRoot.children.find(c => c.friendlyName === 'Tables');
+			stub.restore();
```

Comment:
Can we have stub.restore in "afterEach" or "beforeEach" instead of doing it specifically in some test functions?

Reply:
We can (and do). The reason this is necessary here is because we want to reset the stub for each iteration of the loop.

Note that normally you could use reset() to just reset the stub back to its original state (but still stubbed). But doing that would involve passing the stub around and in the end I just didn't see it being worth the time - making a new stub each time is pretty painless.